Logic/igtlioConnector.cxx

```diff
@@ -223,23 +223,12 @@ int igtlioConnector::Start()
 int igtlioConnector::Stop()
 {
   // Check if thread exists
-  if (this->ThreadID >= 0)
+  if (this->ConnectionThreadID >= 0)
   {
     // NOTE: Thread should be killed by activating ServerStopFlag.
     this->ServerStopFlag = true;
-    {
-      igtlioLockGuard<vtkMutexLock> lock(this->Mutex);
-      for (std::vector<Client>::iterator clientIt = this->Sockets.begin(); clientIt != this->Sockets.end(); ++clientIt)
-        {
-        if (clientIt->Socket.IsNotNull())
-          {
-          clientIt->Socket->CloseSocket();
-          }
-        }
-      this->Sockets.clear();
-    }
-    this->Thread->TerminateThread(this->ThreadID);
-    this->ThreadID = -1;
+    this->Thread->TerminateThread(this->ConnectionThreadID);
```

Comment:
TerminateThread must be avoided (it may be used as a last resort, for recovering from unexpected failures), as it leaves the application in an undeterministic state (we don't know exactly what the thread was doing when it was aborted). Often the method is not available (has no action; does not actually kill the thread).

Instead, a flag should be set to indicate that the thread should stop, and we should wait until the thread actually stops (indicated by setting another flag as the very last statement of the thread function).